Explicit dependency of revision tag in Launcher is required

Issue found while trying to build dynawo-algorithms inside the Continuous Integration docker image (`dynawo/dynawo-distribution-cxx11:latest`). This environment uses `cmake` version 3.12.0. The issue is not reproduced in the Dynawo developer docker image, that contains `cmake` version 3.14.5.

The error is raised when building the `Launcher` sources folder:
`/.../dynawo-algorithms/sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:58:38: fatal error: ../gitversion_algorithms.h: No such file or directory`

The `gitversion_algorithms.h` file is created by the target `revisiontag` defined as a custom target in the `sources/CMakeLists.txt`. It is explicitly stated that the main executable depends on this target.

But the Launcher library also should contain an explicit dependency of the `revisiontag`.

Previously, this error was not raised because dynawo-algorithms was built using make with multiple jobs running in parallel (`make -j $DYNAWO_NB_PROCESSORS_USED`). The file was already built by the job processing dependencies on the main executable when the `Launcher` modules were being compiled. A recent change disabled this parallel build, exposing the issue.

--- a/sources/Launcher/CMakeLists.txt
+++ b/sources/Launcher/CMakeLists.txt
@@ -30,6 +30,7 @@ set(DYN_ALGO_LAUNCHER_HEADERS
   )
 
 add_library(dynawo_algorithms_Launcher SHARED ${DYN_ALGO_LAUNCHER_SOURCES})
+add_dependencies(dynawo_algorithms_Launcher revisiontag)
 
 install(FILES ${DYN_ALGO_LAUNCHER_HEADERS} DESTINATION ${INCLUDEDIR_NAME})
 

